### ISSUE TITLE
Ability to set environment/configuration values at runtime instead of buildtime

### ISSUE THREAD

**Issue by selangley-wa** [angular/angular-cli#28660]
### Description

Angular should provide a common and supported mechanism for the setting of configuration values at runtime (or at least start up time).

Currently, there are multiple implementations / workarounds for this problem. 

**Can we get well-supported mechanism maintained by the Angular Team?**

[Multiple releases using the same, but configurable, Angular Standalone Application build](https://timdeschryver.dev/blog/multiple-releases-using-the-same-but-configurable-angular-standalone-application-build#)

[Compile-time vs. Runtime configuration of your Angular App](https://juri.dev/blog/2018/01/ng-app-runtime-config/)

[Angular, Docker & Nginx: Build once, Deploy anywhere.](https://medium.com/@fabiozuin/build-once-deploy-anywhere-with-angular-17-bf477c49668f)

https://www.npmjs.com/package/@ngx-env/builder

https://www.npmjs.com/package/angular-runtime-config

https://www.npmjs.com/package/@igorissen/angular-env-builder

There was a request: 

https://github.com/angular/angular/issues/52994
ability to use/change environment files after build

to allow the setting of environment-level values at runtime instead of build time.  I believe that was mistakenly marked as a duplicate of:

https://github.com/angular/angular-cli/issues/4318
Request: Method to pass environment variables during build vs file.

### Proposed solution

Include one of the existing user-created implementations described above in Angular with suitable documentation.

Or, at least, pick one of the above and link to it from the Angular documentation.

### Alternatives considered

Use one of the existing user-created implementations.

**Comment by alan-agius4** [comment]
I agree with @dgp1130's point, especially about the reasons for avoiding `process.env`. However, a minor clarification: if the HTTP request is made using Angular's HTTP client, the data is automatically stored in the transfer state, so no additional optimization is necessary.

**Comment by dgp1130** [comment]
We've had some more discussion about this and think it's something which is worth considering at least. `provideRuntimeConfig` is something which _can_ make sense in a CSR app (though I personally think it's much more valuable in an SSR app). However it's probably more of a core framework problem than a tooling problem, so I'll move this out of CLI. I think this is just something we'll need to prioritize based on user interest. So if this is a valuable use case for you, please consider upvoting the issue and/or sharing your particular use case.
